pytorch_lightning/trainer/trainer.py

```diff
@@ -537,6 +537,7 @@ def validate(
         self.state.fn = TrainerFn.VALIDATING
         self.state.status = TrainerStatus.RUNNING
         self.validating = True
+        self.validated_ckpt_path: Optional[str] = None
```

Comment:
These are already defined in `EvaluationLoop.on_trainer_init`.

Also would be nice for you to split the typing fixes into a separate PR